PyTorch added a new keyword argument to nn.Module.named_buffers

[This test failure](https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=211753&view=logs&j=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&t=97651ec4-0b0f-5455-bbb5-3c30427a0a7e&l=13949) happens specifically on torch nightly. This is caused by [this recent commit on PyTorch](https://github.com/pytorch/pytorch/commit/8707c6dfacaed293ddc40cbb5ecf5841568df0e6), which passes a new keyword argument `persistent` to `nn.Module.named_buffers` (overridden by `ThunderModule`).

```py
self = ThunderModule(
  (_model): GPT(
    (transformer): ModuleDict(
      (wte): Embedding(50304, 64)
      (wpe): Embeddin...  )
      )
      (ln_f): LayerNorm()
    )
    (lm_head): Linear(in_features=64, out_features=50304, bias=False)
  )
)
recurse = True

    def buffers(
        self, recurse: bool = True, *, persistent: Optional[bool] = None
    ) -> Iterator[Tensor]:
        r"""Return an iterator over module buffers.
        ...
        """
>       for _, buf in self.named_buffers(recurse=recurse, persistent=persistent):
E       TypeError: ThunderModule.named_buffers() got an unexpected keyword argument 'persistent'

/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py:2682: TypeError
```

--- a/thunder/core/module.py
+++ b/thunder/core/module.py
@@ -90,9 +90,14 @@ class ThunderModule(pytorch.nn.Module):
         )
 
     def named_buffers(self, prefix="", recurse=True, remove_duplicate=True, *, persistent=None):
+        if persistent is not None:
+            orig_buffers = self._model.named_buffers(remove_duplicate=remove_duplicate, persistent=persistent)
+        else:
+            orig_buffers = self._model.named_buffers(remove_duplicate=remove_duplicate)
+
         yield from self._named_parameters_or_buffers(
             self._overrides_buffers,
-            self._model.named_buffers(remove_duplicate=remove_duplicate, persistent=persistent),
+            orig_buffers,
             prefix=prefix,
             recurse=recurse,
             remove_duplicate=remove_duplicate,

--- a/thunder/core/module.py
+++ b/thunder/core/module.py
@@ -64,7 +64,7 @@ class ThunderModule(pytorch.nn.Module):
     def _named_parameters_or_buffers(self, overrides, orig_iter, prefix="", recurse=True, remove_duplicate=True):
         seen_ids = set()
         seen_names = set()
-        for k, v in itertools.chain(overrides.items(), orig_iter(remove_duplicate=remove_duplicate)):
+        for k, v in itertools.chain(overrides.items(), orig_iter):
             if remove_duplicate:
                 id_v = id(v)
                 if id_v in seen_ids:
@@ -83,16 +83,16 @@ class ThunderModule(pytorch.nn.Module):
     def named_parameters(self, prefix="", recurse=True, remove_duplicate=True):
         yield from self._named_parameters_or_buffers(
             self._overrides_parameters,
-            self._model.named_parameters,
+            self._model.named_parameters(remove_duplicate=remove_duplicate),
             prefix=prefix,
             recurse=recurse,
             remove_duplicate=remove_duplicate,
         )
 
-    def named_buffers(self, prefix="", recurse=True, remove_duplicate=True):
+    def named_buffers(self, prefix="", recurse=True, remove_duplicate=True, *, persistent=None):
         yield from self._named_parameters_or_buffers(
             self._overrides_buffers,
-            self._model.named_buffers,
+            self._model.named_buffers(remove_duplicate=remove_duplicate, persistent=persistent),
             prefix=prefix,
             recurse=recurse,
             remove_duplicate=remove_duplicate,

